src/iocore/cache/CacheVC.cc

```diff
@@ -162,8 +162,9 @@ int CacheVC::size_to_init = -1;
 
 CacheVC::CacheVC()
 {
-  size_to_init = sizeof(CacheVC) - (size_t) & ((CacheVC *)nullptr)->vio;
-  memset((void *)&vio, 0, size_to_init);
+  // Initialize Region C
+  size_to_init = sizeof(CacheVC) - (size_t) & (static_cast<CacheVC *>(nullptr))->vio;
```

Comment:
I guess this is calculating the size of Region C start from `vio`, any better idea?

https://github.com/apache/trafficserver/blob/32021ff2fbab643a60eb5800b25f2aad99d97752/include/iocore/cache/CacheVC.h#L246-L253

Reply:
I agree. Region C extends starts at `vio` and goes all the way to the end of the struct, so the difference between the total size and the offset of `vio` is the size of the region. But is it safe to use `nullptr` this way? Is that dereference not undefined behavior?

Reply:
Now, I can read this line. I was reading `&` as `Bitwise AND` and confused. If I read it as dereference, it makes sense. We can do `reinterpret_cast` for the address to `size_t`.

Also, I think it's valid as C++

1. `static_cast<CacheVC *>(nullptr)` - it's called `null pointer conversion`

> A [null pointer constant](https://en.cppreference.com/w/cpp/language/pointer#Null_pointers) can be converted to any pointer type, and the result is the null pointer value of that type.
https://en.cppreference.com/w/cpp/language/implicit_conversion#Pointer_conversions

2. `(static_cast<CacheVC *>(nullptr))->vio` - we can access `vio` memeber
3. `&(static_cast<CacheVC *>(nullptr))->vio` - dereference the `vio`, it shows relative address from `0x00` (`nullptr`)